kernel/kernel-api/src/main/java/io/delta/kernel/Transaction.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import java.util.List;
+import java.util.Map;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.types.StructType;
+import io.delta.kernel.utils.*;
+
+/**
+ * Represents a transaction to mutate a Delta table.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public interface Transaction {
+    /**
+     * Get the schema of the data that is being written to the table.
+     */
+    StructType getSchema(Engine engine);
```

Comment:
The `append` vs `insert` shouldn't matter here. We created a transaction and the transaction expects the data of this schema. Adding `append` or `insert` will complicate the API. Let me know if think otherwise.

Reply:
Is "schema of the data that is being written to the table" = "table schema"?

Reply:
Thats correct. Let me clarify it a bit

Reply:
updated it.

Reply:
Awesome thank you